scripts/ai/tactical/behaviors/ai_rf_hold_steady.nut

```diff
@@ -36,54 +36,51 @@ this.ai_rf_hold_steady <- ::inherit("scripts/ai/tactical/behavior", {
 
 		score = score * this.getFatigueScoreMult(this.m.Skill);
 		local myTile = _entity.getTile();
-		local allies = ::Tactical.Entities.getInstancesOfFaction(_entity.getFaction());
-		local enemies = ::Tactical.Entities.getHostileActors(_entity.getFaction());
 		local useScore = 0.0;
 		local numTargets = 0;
 
-		local relevantAllies = [];		
-
-		foreach( enemy in enemies )
+		foreach (ally in ::Tactical.Entities.getInstancesOfFaction(_entity.getFaction()))
 		{
-			if (!enemy.hasZoneOfControl())
+			local allyTile = ally.getTile();
+			if (allyTile.getDistanceTo(myTile) > 4 || !ally.hasZoneOfControl() || ally.getSkills().hasSkill("effects.rf_hold_steady"))
```

Comment:
Hold Steady grants defensive stats so it shouldnt matter whether that ally emits zone of control. They could have a bow or be sleeping. Both of these scenarios can still receive the buff and profit from it.

Reply:
Technically you are correct - however, for the AI specifically I'd prefer that they use this skill in situations that provide the best bang for the buck. Using it to add a little bit of Defense to ranged units is probably still going to end up in those ranged units dying. So, for the AI specifically I want it to only consider melee units who are engaged with enemy and are in the "fighting" line.